Sources/GDBRemote/PlatformSessionImpl.cpp

```diff
@@ -111,6 +111,11 @@ ErrorCode PlatformSessionImplBase::onLaunchDebugServer(Session &session,
   } else if (GetLogLevel() == kLogLevelPacket) {
     args.push_back("--remote-debug");
   }
+  const char *outputFilename = GetLogOutputFilename();
+  if (outputFilename) {
```

Comment:
Won't this will always be evaluated to true? `GetLogOutputFilename` will always give you a valid, non-null pointer since `sOutputFilename` is just a character array of length `PATH_MAX`. However, the actual contents of it could be {'\0', '\0', ..., '\0'} if it's not initialized. I don't think this is what you want.

Maybe something like:
```
if (strnlen(outputFilename, PATH_MAX) != 0)
```

Reply:
Good catch by @xiaobai. Your storage for the output file name is a statically allocated `char` array, so you'll never return a `NULL` pointer from `GetLogOutputFilename()`. If you switch to using `std::string` in `Log.cpp`, you can just check for `GetLogOutputFilename().length() > 0`.